### ISSUE TITLE
How does XSD/XML association work?

### ISSUE THREAD

**Issue by bergamin** [redhat-developer/vscode-xml#590]
I want to validate an XML against an XSD, but the instructions in `Validation.md` aren't clear. It just states the following:

> You can `validate your XML with a grammar` (XSD, DTD). In order to do so, you must associate your XML document with a grammar. Once you have done this, you will benefit from: [...]

But it doesn't explain how the association works. I've seen somewhere that both XSD and XML must be in the same folder, but that can't be enough because the XSD schema has multiple files. I would expect a "validate" button somewhere or right clicking on the file on the explorer to choose the root XSD file to validate that XML against (or select both files and then right click them to a "Validate XML" option)

Or maybe I understood this extension wrong and am expecting this extension to do something it doesn't... I'm not the creator of the XSD. The XML is automatically generated by a system and one file got rejected by an automatic validator that doesn't say what's wrong. I just want to validate this file to see what node caused the problem and why.

Thanks

**Comment by fbricon** [comment]
You can configure file associations: https://github.com/redhat-developer/vscode-xml/blob/master/docs/Validation.md#XML-file-association-with-xsd if you don't want to modify your files

**Comment by angelozerr** [comment]
> I would expect a "validate" button somewhere or right clicking on the file on the explorer to choose the root XSD file to validate that XML against 

I suggest you enable codelens and after that you will have this feature:

![BindToGrammarWithCodeLensDemo](https://user-images.githubusercontent.com/1932211/131639682-0bfcb7b4-7997-4696-8ee4-384e33b13356.gif)

See more 
https://github.com/redhat-developer/vscode-xml/blob/master/docs/BindingWithGrammar.md#codelens

But as you wish to use `File Association` binding kind, the vscode-xml 0.18.0 (released) will not help you with codelens binding, because it doesn't provide the capability to bind with file association. BUT if you download and install last vsix at https://download.jboss.org/jbosstools/vscode-xml/staging/?C=M;O=D, you will benefit with this feature 

![FileAssociationCodeLensDemo](https://user-images.githubusercontent.com/1932211/131640684-ff93ef70-82e2-4f7e-94a2-3f174f2f32a9.gif)

**Comment by bergamin** [comment]
Although the instructions now seem clear to me, I'm struggling with an error message, so I'm reopening this issue. If anybody could help... I like the idea of using a pattern so I wouldn't need to edit the XML files, so I went this route:

![image](https://user-images.githubusercontent.com/14339291/131753906-9d9c8de4-690f-46d6-9c2d-a407a1893923.png)

Where "systemId" has the path to the XSD file. The following error shows up in the XML file:

![image](https://user-images.githubusercontent.com/14339291/131754256-7c4ec1d8-485e-4942-a6e8-48f26decb2da.png)

Where "file:///" has the path to the XML file itself (it broke on accents and letters that aren't present in English, but I thin kthat's just because it's an URL). The tag being hidden by the message is the XML definition (`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>`)

It also shows on another XML that doesn't match the pattern so I guess my problem is the extension not understanding the pattern I typed there for applying the XSD schema.

The filename follows this pattern: `99999999_99999999_dados_publicos.xml`. I checked the namespace and it's correct too.

What could this be?

Thanks

**Comment by angelozerr** [comment]
Could you share please your workspace folder which conatins:

 * your settings.json
 * your XML
 * your XSD

**Comment by angelozerr** [comment]
If your xml is not bound its a pattern problem. @rgrunber should implement (when jhe will havetime) pattern validator which will validate the pattern when you will use the file association kind with codelens. I think its a feature that it should help you to write pattern

**Comment by bergamin** [comment]
@angelozerr, unfortunately I can't share the XML as it has client personal data, but now that you mention it, I looked back to the gifs you posted earlier. The pattern you used is different from the one in the example. I used `*` to get the whole path to the file and that was the issue. Changing the pattern based on yours, worked: `**/*_dados_publicos.xml`. Now I can see the issues without editing the files:

![image](https://user-images.githubusercontent.com/14339291/131838040-bfecbf43-d2f0-4acd-827d-0de3883cde40.png)

Thanks for the help!
